sdk/pinocchio/src/sysvars/instructions.rs

```diff
@@ -0,0 +1,287 @@
+use crate::{
+    account_info::AccountInfo, instruction::AccountMeta, program_error::ProgramError,
+    pubkey::Pubkey, sanitize_error::SanitizeError,
+};
+
+use core::mem::size_of;
+
+/// Sysvar1nstructions1111111111111111111111111
+pub const INSTRUCTIONS_ID: Pubkey = [
+    0x06, 0xa7, 0xd5, 0x17, 0x18, 0x7b, 0xd1, 0x66, 
+    0x35, 0xda, 0xd4, 0x04, 0x55, 0xfd, 0xc2, 0xc0,
+    0xc1, 0x24, 0xc6, 0x8f, 0x21, 0x56, 0x75, 0xa5, 
+    0xdb, 0xba, 0xcb, 0x5f, 0x08, 0x00, 0x00, 0x00,
+];
+
+pub struct Instructions();
+
+/// Load the current `Instruction`'s index in the currently executing
+/// `Transaction`.
+///
+/// `data` is the instructions sysvar account data.
+///
+/// # Safety
+///
+/// This function is unsafe because it does not verify the address of the sysvar account.
+/// 
+/// It is typically used internally within the `load_current_index_checked` function, which
+/// performs the necessary address verification. However, to optimize performance for users
+/// who have already conducted this check elsewhere in their code, we have exposed it as an
+/// unsafe function rather than making it private, as was done in the original implementation.
+#[inline(always)]
+pub unsafe fn load_current_index(data: &[u8]) -> u16 {
+    let len = data.len();
+    *(data.as_ptr().add(len - 2) as *const u16)
```

Comment:
We will need to use `u16::from_le_bytes` here since we cannot guarantee that the alignment will be correct.